assets/src/js/admin/reports/components/list/index.js

```diff
@@ -1,6 +1,35 @@
-const List = (props) => {
+import { useEffect, createRef } from 'react'
+import './style.scss';
+
+const List = ({onScrollEnd, children}) => {
+
+    const list = createRef()
+
+    useEffect(() => {
+
+        function checkScroll (evt) {
+
+            const remaining = evt.target.scrollHeight - evt.target.scrollTop
+            const height = evt.target.offsetHeight
+
+            if (remaining <= height) {
+                onScrollEnd()
+            }
+
+        }
+        
+        list.current.addEventListener('scroll', checkScroll)
+    
+        return function cleanup () {
+            list.current.removeEventListener('scroll', checkScroll)
+        }
+
+    }, [])
+
     return (
-        <h1>List goes here!</h1>
+        <div ref={list} class='list'>
+            {children}
+        </div>
     )
 }
 export default List
```

Comment:
Please add `List.propTypes` and `List.defaultProps`.

Reply:
done! also added styles for "default-item" in the event that a list is empty, to display an error message.

Reply:
If it's a required element, just mark the prop type as required: `PropTypes.node.isRequired`. As this error shouldn't ever really display as it's not a runtime error, but a failure to utilize this component properly.